core/iwasm/common/wasm_exec_env.c

```diff
@@ -208,10 +208,20 @@ void
 wasm_exec_env_set_thread_info(WASMExecEnv *exec_env)
 {
     uint8 *stack_boundary = os_thread_get_stack_boundary();
+
+#if WASM_ENABLE_THREAD_MGR != 0
+    WASMCluster *cluster = wasm_exec_env_get_cluster(exec_env);
+    if (cluster)
+        os_mutex_lock(&cluster->lock);
```

Comment:
Shall we add lock for these operations? It seems these field are only read later by this thread.

Reply:
These two are the operations manipulating it in different threads
https://github.com/bytecodealliance/wasm-micro-runtime/blob/38c67b3f4815a628ce0f9bc22186bface4678089/core/iwasm/libraries/thread-mgr/thread_manager.c#L858

https://github.com/bytecodealliance/wasm-micro-runtime/blob/38c67b3f4815a628ce0f9bc22186bface4678089/core/iwasm/common/wasm_exec_env.c#L211

Reply:
OK, could we use `exec_env->wait_lock` to lock it in the two places?

Reply:
Yes, pushed a commit with the change, it makes more sense to use the more specific/fine-grained lock in the exec_env instead of the cluster one.

I tested the change in classic interpreter, fast interpreter and aot modes running the tests in #1985 with the thread sanitizer and no data race is reported anymore.